### PR TITLE
MBSA-724 Extract di logic to DI

### PR DESCRIPTION
**Pull request by sergioboi**

### CODE REVIEW

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/rest/HttpServer.kt [hunk 1]
```diff
@@ -14,12 +14,15 @@ import io.ktor.server.routing.routing
 
 internal class HttpServer(
     private val handlers: List<RequestHandler<*>>,
-    private val debug: Boolean
+    private val debug: Boolean,
+    private val port: Int,
+    private val gracePeriodMs: Long = 500,
```
Comment (RuslanMingaliev) [inline subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/rest/HttpServer.kt:19]: we don't need that

File: subprojects/emcee/worker/src/test/kotlin/com/avito/emcee/worker/rest/HttpServerTest.kt [hunk 2]
```diff
@@ -20,15 +20,24 @@ import java.net.ServerSocket
...
-    private var port: Int? = null
+    private companion object {
+        private fun findAvailablePort(): Int {
+            val s = ServerSocket(0)
+            val port = s.localPort
+            s.close()
+            return port
+        }
+
+        private val port: Int = findAvailablePort()
```
Comment (RuslanMingaliev) [inline subprojects/emcee/worker/src/test/kotlin/com/avito/emcee/worker/rest/HttpServerTest.kt]: It should be found on each call, for each test. Not only once.

File: subprojects/emcee/worker/src/test/kotlin/com/avito/emcee/worker/rest/HttpServerTest.kt [hunk 3]
```diff
@@ -56,15 +65,11 @@ class HttpServerTest {
         server.stop()
     }
 
-    private fun findAvailablePort(): Int {
-        val s = ServerSocket(0)
-        val port = s.localPort
-        s.close()
-        return port
+    private fun createServer(): HttpServer {
+        return HttpServer(
```
Comment (RuslanMingaliev) [inline subprojects/emcee/worker/src/test/kotlin/com/avito/emcee/worker/rest/HttpServerTest.kt:58]: it looks more pretty with builder pattern for me

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/di/WorkerDI.kt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import com.avito.emcee.worker.internal.rest.handler.ProcessingBucketsRequestHandler
+import com.avito.emcee.worker.internal.storage.ProcessingBucketsStorage
+import com.avito.emcee.worker.internal.storage.SingleElementProcessingBucketsStorage
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import kotlin.time.ExperimentalTime
+
+@ExperimentalTime
+internal class WorkerDI(
```
Comment (RuslanMingaliev) [inline subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/di/WorkerDI.kt:24]: dagger?

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/di/WorkerDI.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    private val config: Config,
+    private val debugMode: Boolean
+) {
+
+    private val workerIdProvider: WorkerIdProvider = HostnameWorkerIdProvider()
+    private val okHttpClient = OkHttpClient.Builder().apply {
+        if (debugMode) {
+            addInterceptor(HttpLoggingInterceptor { message ->
+                println(message)
```
Comment (RuslanMingaliev) [inline subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/di/WorkerDI.kt]: Logger should be used here too

**Review by RuslanMingaliev — APPROVED** [review]

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/rest/HttpServer.kt (hunk 1 above)
Comment (sergioboi) [reply subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/rest/HttpServer.kt:19]: I noticed that after failure our worker works for minute more. So I decide to use this properties explicitly
But problem was not in stopping HttpServer

File: subprojects/emcee/worker/src/test/kotlin/com/avito/emcee/worker/rest/HttpServerTest.kt (hunk 2 above)
Comment (sergioboi) [reply subprojects/emcee/worker/src/test/kotlin/com/avito/emcee/worker/rest/HttpServerTest.kt]: Before in test was @BeforeAll annotation I think that means that we make it one time before tests
What the point each time find availablePorts

File: subprojects/emcee/worker/src/test/kotlin/com/avito/emcee/worker/rest/HttpServerTest.kt (hunk 3 above)
Comment (sergioboi) [reply subprojects/emcee/worker/src/test/kotlin/com/avito/emcee/worker/rest/HttpServerTest.kt:58]: I against the builder in current case. In my opinion builder needs when you have many parameters and many configurations of object and you make them dynamically in runtime. It's not about current case

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/di/WorkerDI.kt (hunk 4 above)
Comment (sergioboi) [reply subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/di/WorkerDI.kt:24]: Let's stay without dagger. Maybe in the future

**Pull request merged**